ckanext/scheming/plugins.py

```diff
@@ -331,13 +331,18 @@ def expand_form_composite(data, fieldnames):
     fieldnames -= set(data)
     if not fieldnames:
         return
+    indexes = []
     for key in sorted(data):
         if '-' not in key:
             continue
         parts = key.split('-')
         if parts[0] not in fieldnames:
             continue
+        if parts[1] not in indexes:
+            indexes.append(parts[1])
         comp = data.setdefault(parts[0], [])
+        # if status index is missing, replace with next status
```

Comment:
this code applies to any composite repeating group of fields, status is just the one we're using in our theme

Reply:
I can remove this comment